### DISCUSSION TITLE
Enable HTTPS in the Web Client Beta

### DISCUSSION THREAD

**Discussion by AdanC1b3r** [rustdesk/rustdesk#5145] [Q&A]
Add the security and privacy benefits of an HTTPS certificate, although it is still in beta mode, it is a good practice that should not be missed.

**Comment by AdanC1b3r** [comment]
They can use Let's Encrypt.

Let's Encrypt is a free, automated, and open certificate authority brought to you by the nonprofit Internet Security Research Group (ISRG).

**Reply by dinger1986** [reply]
for pro for the web console there is a nginx proxy in front which uses letsencrypt, it would need the same for the web client

**Comment by TBG-FR** [comment]
I'm really not sure they'll add HTTPS to the Web Client, see #856 **Anyway the web client is not maintained, not enough resources to work on it, and it's a real mess to build/use in it's today's state...**

RustDesk Server does no support WSS (and will not...), so if you use a HTTPS web client, the browser will block the connection (HTTPS -> WS is not allowed)... (cf. https://github.com/rustdesk/rustdesk-server/issues/337 and https://github.com/rustdesk/rustdesk-server/issues/94 )

Only thing you can do is setup some proxy with https, maybe (ex nginx). **The best thing would be the server supporting WSS, or you putting a WS to WSS proxy**, because you can tell the web client to use WSS, it works

![rustdesk](https://github.com/rustdesk/rustdesk/assets/18146038/ca8b621c-f19a-4ab4-b962-37ecc2f6c8b4)

**Reply by TBG-FR** [reply]
Managed to achieve the last case of the above schema 🤓 https://github.com/rustdesk/rustdesk/discussions/6023#discussioncomment-8687562

**Comment by rustdesk** [comment]
Update about web client: https://twitter.com/rustdesk/status/1771454382426448375
